### PR TITLE
Bookstore: Add Reducers And Actions

### PR DESCRIPTION
**Pull request by AliShahSafdari**
This is PR add features:

- Add Redux Toolkit.
- Configure a Redux store.
- Define slice for Books and Categories.
- Add reducer for add-book and remove-book.

### CODE REVIEW

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  categories: [],
+};
+
+const categorySlice = createSlice({
+  name: 'Catagories',
+  initialState,
+  status: 'Under construction',
+  reducers: {
+    checkStatus: initialState,
+  },
```
Comment (souad988) [inline src/redux/categories/categoriesSlice.js]: Great work done so far 👍🏼  , but this slice was not correctly configured , the idea from this statement in the requirements  Includes a reducer that checks the status and always returns "Under construction" (the initial state should check to that string)  is that the action payload  that the reducer checks and return a value accordingly

`initialState: {
    categories: [],
  },
  reducers: {
    checkStatus: (state, action) => {
      state.categories =
        action.payload === "Under construction"
          ? "Under construction"
          : state.categories
    },
  },
`

**Review by souad988 — CHANGES_REQUESTED** [review]
Hi @AliShahSafdari  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


242722065-db428b0c-8dfc-49c4-a156-42f227e32868


## Highlights
Good PR description
Professional README
Use redux toolkit
Good file structure

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js (hunk 1 above)
Comment (AliShahSafdari) [reply src/redux/categories/categoriesSlice.js]: Thanks

**Review by nedjwab — APPROVED** [review]
# 🟢 Project Approved! 🏆

## Hi  @AliShahSafdari  👋

Your project is complete! ✔️✔ , I loved your work, it looks great 🔥🔥🔥🔥 🏁

Your code was very professional 💯

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉

 
## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors

- The code is top-notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**